onnx/test/optimizer_test.py

```diff
@@ -1120,6 +1120,34 @@ def test_eliminate_nop_dropout(self):  # type: () -> None
                 assert len(optimized_model.graph.node) == 1
                 assert optimized_model.graph.node[0].op_type == "Log"
 
+    def test_fuse_reduction_unsqueeze(self):  # type: () -> None
+        for reduction in ["ReduceL1", "ReduceL2", "ReduceLogSum",
+                          "ReduceLogSumExp", "ReduceMax", "ReduceMean",
+                          "ReduceMin", "ReduceProd", "ReduceSum", "ReduceSumSquare"]:
+            for axes1 in [[1], [1, 2]]:
+                for axes2 in [[1], [1, 2]]:
+                    for keepdim in [False, True]:
+                        node = helper.make_node(reduction, ["X"], ["Y"], axes=axes1, keepdims=keepdim)
+                        node1 = helper.make_node("Unsqueeze", ["Y"], ["Z"], axes=axes2)
+                        graph = helper.make_graph(
+                            [node, node1],
+                            "test",
+                            [helper.make_tensor_value_info(
+                                "X", TensorProto.FLOAT, (5, 7, 9))],
+                            [helper.make_tensor_value_info("Z", TensorProto.FLOAT, (5, 9))])
```

Comment:
output shape are not valid in some cases

Reply:
fixed.